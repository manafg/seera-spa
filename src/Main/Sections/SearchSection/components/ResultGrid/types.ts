type ResultGridProps = {
    clearSearch : ()=>void;
};

export type {ResultGridProps}
