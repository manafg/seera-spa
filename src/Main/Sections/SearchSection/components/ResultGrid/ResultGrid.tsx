import { SimpleGrid, Box, Center, Button } from "@chakra-ui/react";
import React, { useContext } from "react";

import Card from "../../../../Components/Card";
import { DealStateContext } from "../../../../Context/DealsContext";
import { Deal } from "../../../../Context/DealsContext/type";
import { ResultGridProps } from "./types";

function ResultGrid({ clearSearch }: ResultGridProps) {
  const state = useContext(DealStateContext);
  const deals: any = state[0];

  return (
    <>
      {deals.length ? (
        <SimpleGrid columns={2} spacing={10}>
          {deals?.map((item: Deal, index: number) => (
            <Box key={index}>
              <Card
                name={item.name}
                price={item.price}
                city={item.city}
                available_on={item.available_on}
              />
            </Box>
          ))}
        </SimpleGrid>
      ) : (
        <Center bg='#F7F7F7' mt='200px' ml='300px' color='black'>
          No Result match your search
          <Button p='3' colorScheme="red" variant="link" onClick={clearSearch}>
             Reset
          </Button>
        </Center>
      )}
    </>
  );
}

export default ResultGrid;
