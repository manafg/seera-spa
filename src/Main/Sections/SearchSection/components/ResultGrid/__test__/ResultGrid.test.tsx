import { render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";

import ResultGrid from "../ResultGrid";

import "@testing-library/jest-dom/extend-expect";
import rrender from "react-test-renderer";

describe("Snapshot tests", () => {
  test("Matches existing Snapshot", () => {
    // @ts-ignore
    const tree = rrender.create(<ResultGrid />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
