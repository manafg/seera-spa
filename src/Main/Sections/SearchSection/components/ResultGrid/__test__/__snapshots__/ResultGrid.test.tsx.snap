// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Matches existing Snapshot 1`] = `
<div
  className="css-kkkcrq"
>
  No Result match your search
  <button
    className="chakra-button css-1fhw8o5"
    type="button"
  >
    Reset
  </button>
</div>
`;
